pcsx2/GameDatabase.cpp

```diff
@@ -182,68 +235,129 @@ GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::entryFromYaml(const std::str
 	return gameEntry;
 }
 
-GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::findGame(const std::string serial)
+// NOTE - when we load more files relative from the .exe, we'll want to isolate this
+// to insulate the GameDB (and others) from the GUI.  At the moment though, this is the only file.
+std::optional<wxString> getFileAsStream()
 {
-	std::string serialLower = strToLower(serial);
-	Console.WriteLn(fmt::format("[GameDB] Searching for '{}' in GameDB", serialLower));
-	if (gameDb.count(serialLower) == 1)
+	// TODO - config - refactor with std::filesystem/ghc::filesystem
+	wxString file("GameIndex.yaml");
+	if (wxFileName(file).IsRelative())
 	{
-		Console.WriteLn(fmt::format("[GameDB] Found '{}' in GameDB", serialLower));
-		return gameDb[serialLower];
+		// InstallFolder is the preferred base directory for the DB file, but the registry can point to previous
+		// installs if uninstall wasn't done properly.
+		// Since the games DB file is considered part of pcsx2.exe itself, look for it at the exe folder
+		//   regardless of any other settings.
+
+		// Note 1: Portable setup didn't suffer from this as install folder pointed already to the exe folder in portable.
+		// Note 2: Other folders are either configurable (memcards, etc) or create their content automatically (inis)
+		//           So the games DB was really the only one that suffers from residues of prior installs.
+
+		//wxDirName dir = InstallFolder;
+		wxDirName dir = (wxDirName)wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath();
+		file = (dir + file).GetFullPath();
 	}
 
-	Console.Error(fmt::format("[GameDB] Could not find '{}' in GameDB", serialLower));
-	GameDatabaseSchema::GameEntry entry;
-	entry.isValid = false;
-	return entry;
+	if (!wxFileExists(file))
+	{
+		Console.Error(L"[GameDB] Database Not Found! [%s]", WX_STR(file));
+		return std::nullopt;
+	}
+	return file;
 }
 
-int YamlGameDatabaseImpl::numGames()
+std::ifstream getFileStream(wxString path)
 {
-	return gameDb.size();
+#ifdef _WIN32
+	return std::ifstream(path.wc_str());
+#else
+	return std::ifstream(path.c_str());
+#endif
 }
 
-bool YamlGameDatabaseImpl::initDatabase(std::ifstream& stream)
+void YamlGameDatabaseImpl::initDatabase()
 {
+	const ryml::Callbacks preserve_callbacks = ryml::get_callbacks();
+	const c4::error_callback_type preserve_c4_callback = c4::get_error_callback();
+	auto callbacks = ryml::get_callbacks();
```

Comment:
This is not actually set anywhere. The only `ryml::set_callbacks` is at the bottom, is it correct?

Reply:
The library has default callbacks set from my understanding, which is why i preserve and restore them after im finished here -- `auto callbacks = ryml::get_callbacks();` is a reference which i modify the error handler on the next line.